core/shared/platform/rt-thread/rtt_platform.c

```diff
@@ -0,0 +1,173 @@
+//
+// Created by alvis on 2020/12/14.
+//
+
+#include <platform_api_vmcore.h>
+#include <platform_api_extension.h>
+
+typedef struct os_malloc_list {
+    void* real;
+    void* used;
+    rt_list_t node;
+}os_malloc_list_t;
+
+static rt_list_t mem_list;
+
+int bh_platform_init(void)
+{
+    rt_list_init(&mem_list);
+    return 0;
+}
+
+void bh_platform_destroy(void)
+{
+    while(!rt_list_isempty(&mem_list))
+    {
+        os_malloc_list_t *ptr = rt_list_entry(mem_list.next, os_malloc_list_t, node);
+        rt_list_remove(&ptr->node);
+        rt_free(ptr->real);
+        rt_free(ptr);
+    }
+}
+
+
+void *os_malloc(unsigned size)
+{
+    void* buf = rt_malloc(size + 8);
+    if (((rt_ubase_t)buf)&7)
```

Comment:
Not sure why here it doesn't check buf != NULL, but check (buf & 7), could buf's lowest 3 bits be zero, and higher bits be not zero? e.g. 0x11110, or 0x11118.

Reply:
wamr need heap init struct be aligned to 8-bytes, otherwise iwasm will print error information: `[GC_ERROR]heap init struct buf not 8-byte aligned`. these codes are used to make sure the memories allocated are all align to 8-byte.

Reply:
Got it, thanks a lot! And should we also check the input pointer ptr like os_free in os_realloc? The ptr might be not the original buffer returned by rt_malloc.

Reply:
there is a list to save the map from fixed-address to origin-address named `mem_list`. if the address of memory allocated is not align to 8-byte, this address and fixed-address will be save into this list. while `os_free` be called, will also check this list to find the original address of the pointer pass in.

as for `os_realloc`, I have not done anything yet, I will resolve it later.

Reply:
Hi, it seems that the performance might be not so good if saving the map to mem_list, e.g. in os_free, we need to traverse the list. And also for multi-thread, normally we should to add lock for the list. How about allocating memory with extra 16 bytes, saving the original address to the next 8-byte aligned address, and return the next next 8-byte aligned address? e.g.:
void *os_malloc(unsigned size)
{
    uint8 *buf = rt_malloc(size + 16);
    uint8 *buf_aligned;

    if (buf) {
        buf_aligned = (uint8 *)(((uintptr_t)buf + 7) & ~(uintptr_t)7);
        *(uint8 **)buf_aligned = buf;
        buf = buf_aligned + 8;
    }
    return buf;
}

void os_free(void *ptr)
{
    uint8 *buf_aligned;

    if (ptr) {
        buf_aligned = (uint8 *)ptr - 8;
        buf = *(uint8 **)buf_aligned;
        rt_free(buf);
    }
}

Reply:
thinks for your reply and advice, It seems be a good idea, I will try this today.

Reply:
I have rewrite these three method, please check.